src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheckTest.java

```diff
@@ -157,4 +157,36 @@ public void testInvalidOption() throws Exception {
                             + "Cannot set property 'option' to 'invalid_option' in module"));
         }
     }
+
+    @Test
+    public void testAllowEmptyCaseWithText() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(EmptyBlockCheck.class);
+        checkConfig.addAttribute("option", BlockOption.TEXT.toString());
+        checkConfig.addAttribute("tokens", "LITERAL_CASE");
+        final String[] expected = {
+            "12:21: " + getCheckMessage(MSG_KEY_BLOCK_EMPTY, "case"),
+            "17:13: " + getCheckMessage(MSG_KEY_BLOCK_EMPTY, "case"),
+            "29:21: " + getCheckMessage(MSG_KEY_BLOCK_EMPTY, "case"),
+            "31:29: " + getCheckMessage(MSG_KEY_BLOCK_EMPTY, "case"),
+            "32:21: " + getCheckMessage(MSG_KEY_BLOCK_EMPTY, "case"),
+        };
+        verify(checkConfig, getPath("InputEmptyCase.java"), expected);
+    }
+
+    @Test
+    public void testEmptyCase() throws Exception {
```

Comment:
you do not test all cases, you set the block option to STMT. It should be in the test name.